collect/trace_decision.go

```diff
@@ -66,25 +78,47 @@ func newDroppedTraceDecision(msg string) ([]TraceDecision, error) {
 	return decisions, nil
 }
 
-func newKeptDecisionMessage(tds []TraceDecision) (string, error) {
+func newKeptDecisionMessage(tds []TraceDecision, senderID string) (string, error) {
 	if len(tds) == 0 {
 		return "", fmt.Errorf("no kept trace decisions provided")
 	}
+
+	if senderID == "" {
+		return "", fmt.Errorf("no sender ID provided")
+	}
+
 	compressed, err := compress(tds)
 	if err != nil {
 		return "", err
 	}
-	return string(compressed), nil
+	return senderID + "|" + string(compressed), nil
 }
 
-func newKeptTraceDecision(msg string) ([]TraceDecision, error) {
-	compressed, err := decompressKeptDecisions([]byte(msg))
+func newKeptTraceDecision(msg string, senderID string) ([]TraceDecision, error) {
+	data := strings.SplitN(msg, "|", 2)
+	if len(data) != 2 {
+		return nil, fmt.Errorf("invalid dropped decision message")
+	}
+
+	if data[0] != senderID {
+		return nil, nil
+	}
+
+	compressed, err := decompressKeptDecisions([]byte(data[1]))
 	if err != nil {
 		return nil, err
 	}
 	return compressed, nil
 }
 
+func isMyDecision(msg string, senderID string) bool {
+	data := strings.SplitN(msg, "|", 2)
```

Comment:
It's 4x faster with no allocations if you replace this function with `strings.HasPrefix()`.

Also, the other calls to strings.SplitN(..,..,2) could also be sped up with `strings.IndexRune()` instead. (I just noticed this, we might have other places that would also benefit.)